CarSeatCoverSearch.java

```diff
@@ -0,0 +1,205 @@
+import javax.swing.*;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+
+/* CarSeatCoverSearch.java has a main method that uses the methods getUserDetails(), getUserCriteria(), loadInventory(), writeOrderRequestToFile(), and findMatch() */
+public class CarSeatCoverSearch {
+
+    // fields
+    private final static String appName = "SuperGeek Auto";
+    private final static String filePath = "./inventory.txt";
+    private static Inventory carSeatCovers;
+
+    /**
+     * main method used to interact with the user
+     * @param args not required
+     */
+    public static void main(String[] args) {
+        carSeatCovers = loadInventory();
+
+        JOptionPane.showMessageDialog(null, "Welcome to SuperGeek Auto!\n\tTo start, click OK.", appName, JOptionPane.QUESTION_MESSAGE);
+        CarSeatCover dreamProduct = getUserCriteria();
+        List<CarSeatCover> potentialMatches = carSeatCovers.findMatch(dreamProduct);
+        if(potentialMatches.size()>0){
+            Map<String,CarSeatCover> options = new HashMap<>();
+            StringBuilder infoToShow = new StringBuilder("Matches found!! The following products meet your criteria: \n");
+            for (CarSeatCover potentialMatch : potentialMatches) {
+                infoToShow.append(potentialMatch.formatProductDescription());
+                options.put(potentialMatch.getProductName(), potentialMatch);
+            }
+            String order = (String) JOptionPane.showInputDialog(null,infoToShow+"\n\nPlease select which car seat cover you'd like to order:",appName, JOptionPane.QUESTION_MESSAGE,null,options.keySet().toArray(), "");
+            if(order==null) System.exit(0);
+            else{
+                CarSeatCover chosenCarSeatCover = options.get(order);
+                Geek applicant = getUserDetails();
+                writeOrderRequestToFile(applicant, chosenCarSeatCover, dreamProduct.getAllTypes().iterator().next());
```

Comment:
this `CarSeatCover.getAllTypes().iterator().next()` gets clunky and unwieldy pretty quickly, a good candidate for turning into a method on the `CarSeatCover` class. Maybe something like `CarSeatCover.getFirstType()`. 

something to consider